#!/usr/bin/env python
PACKAGE = "qidian_tactics"  
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 定义参数
gen.add("is_pub", bool_t, 0, "是否发布", False)
gen.add("own_robot_HP", int_t, 0, "Own robot HP", 600, 0, 600)
gen.add("own_base_HP", int_t, 0, "Own base HP", 1500, 0, 1500)
gen.add("enemy_base_HP", int_t, 0, "Enemy base HP", 1500, 0, 1500)
gen.add("event_data", int_t, 0, "Event data", 0, 0, 255)
gen.add("maximum_HP", int_t, 0, "Maximum HP", 600, 0, 600)
gen.add("shoot_num", int_t, 0, "Shoot number", 750, 0, 750)
gen.add("rfid_status", int_t, 0, "RFID status", 0, 0, 255)
gen.add("detect_enemy", bool_t, 0, "Detect enemy", False)
gen.add("Virtual_Shield_Status", double_t, 0, "Virtual Shield Status", 100, 0.0, 100.0)
gen.add("own_supply_area_Status", bool_t, 0, "Own supply area status", False)
gen.add("Central_area_status", bool_t, 0, "Central area status", False)

exit(gen.generate(PACKAGE, "qidian_tactics", "RobotParameters"))
